Add average sensitivity and precision to metrics

jsalbert opened the pull request:
Accuracy = weighted average sensitivity (sum(sensitivity_class) * samples_class / samples)
Sensitivity =  average sensitivity (sum(sensitivity_class) / n_classes)
Precision = average precision (sum(precision_class) / n_classes)
Weighted precision = weighted average precision (sum(precision_class) * samples_class / samples)
F1_score = Accuracy * weighted_precision

adriaromero commented on tests/test_eval.py:
```diff
@@ -159,6 +159,8 @@ def test_show_results(evaluator_mobilenet, test_catdog_dataset_path):
     assert average_df['model'][0] == 'catdog-mobilenet.hdf5'
     assert average_df['accuracy'][0] == 0.75
+    assert average_df['sensitivity'][0] == 0.75
+    assert average_df['weighted_precision'][0] == 0.833
```
(Being really picky here) shall we list `precision` before `weighted_precision`? 
```suggestion
    assert average_df['precision'][0] == 0.833
```

jsalbert replied:
I changed it to accuracy, weighted precision, sensitivity, precision. To reflect first 2 weighted, second 2 without

adriaromero commented on tests/test_eval.py:
```diff
@@ -159,6 +159,8 @@ def test_show_results(evaluator_mobilenet, test_catdog_dataset_path):
     assert average_df['accuracy'][0] == 0.75
+    assert average_df['sensitivity'][0] == 0.75
+    assert average_df['weighted_precision'][0] == 0.833
     assert average_df['precision'][0] == 0.833
```
```suggestion
    assert average_df['weighted_precision'][0] == 0.833
```

Is it a coincidence these two metric values are equal?

jsalbert replied:
when n_samples is the same they will be

jeremykawahara commented on keras_eval/metrics.py:
```diff
@@ -113,15 +118,20 @@ def metrics_top_k(y_probs, y_true, concepts, top_k, round_decimals=7):
 
     metrics['average'] = OrderedDict([('accuracy', accuracy if len(accuracy) > 1 else accuracy[0]),
-                                      ('precision', round(utils.safe_divide(sum(average_precision), total_samples),
+                                      ('sensitivity', round(utils.safe_divide(sum(average_sensitivity), len(concepts)),
```
`average_sensitivity` is the sensitivity computed for each class right? Not the averaged sensitivity.

Would it be clearer to call it something like `sensitivity_per_class`? I think this can be simplified to something like:

`mean_sensitivity = np.mean(sensitivity_per_class)`

jsalbert replied:
I changed the nomenclature, however we were not doing `np.mean` because of the NaN values.

jsalbert commented:
I renamed the variables to avoid misunderstandings (I think) and fixed the order to reflect that the first 2 are weighted and the second 2 are just the average of the individual metrics.
